### PR TITLE
isDM in Notifications

### PR DESCRIPTION
**Pull request by pixlwave**
Fixes DM rendering in notifications, so that a DM must be a direct room with 2 people in.

Also bumps the SDK.

### CODE REVIEW

File: ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift [hunk 1]
```diff
@@ -37,15 +37,31 @@ protocol NotificationItemProxyProtocol {
...
+            default:
+                return false
+            }
+        default:
+            return false
+        }
+    }
+
+    var isDM: Bool {
+        isRoomDirect && roomJoinedMembers <= 2
```
Comment (pixlwave) [inline ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift:63]: Should this also include `isEncrypted`, or (more likely) is there an issue detecting this reliably in notifications?

Comment (Velin92) [reply ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift:63]: we are not really using it so far so... yeah we can remove it for now (also doesn't really work properly mostly for invite notificatons)

Comment (alfogrillo) [reply ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift:63]: Not sure it's correct to use the joined members count here.
In other contexts we use the active members count:
```swift
    var isEncryptedOneToOneRoom: Bool {
        isDirect && isEncrypted && activeMembersCount <= 2
    }
```
(I don't think "encrypted" plays a role, maybe we should remove it everywhere)
wdyt?

**Review by Velin92 — APPROVED** [review]

File: ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift (hunk 1 above)
Comment (alfogrillo) [reply ElementX/Sources/Services/Notification/Proxy/NotificationItemProxy.swift:63]: There are several problems with membership ATM. They seem related with a dirty cache inside the rust sdk.
I tracked this issue here -> https://github.com/matrix-org/matrix-rust-sdk/issues/2078

> Also I think we want to consider a room a DM until a third person joins from a notification point of view.

Ok, then! 👍🏻

**Review by alfogrillo — APPROVED** [review]
LGTM

**Pull request merged**